### PR TITLE
Fixes buggy params for AwsRedirectController

### LINKED ISSUE #199: Downloads not working
getting a 500 on demo site, while development gives a `undefined method [] for nil class`.

It's trying to access `params[:id]` but `params` is nil. Not sure why.

In config/routes.rb there is `resources :downloads, only: [:show]`, which I thought would set up the `#show` route to accept an `:id` param.

### PR DESCRIPTION
**Pull request by afred**
fixes #199

### CODE REVIEW

**Comment by afred** [comment]
@mccalluc please review

NOTE: `params` is a method from `ActionController::Base` that automagically retrieves a hash of http params. Reassigning `params` to a hash causes `params` to no longer be a method. In this case, we were trying to call the method `params` while also assigning a value to `params`, chaos ensues. Essentially, we were trying to do the following...

```ruby
class Foo
  def my_hash
    {alpha: "bravo"}
  end

  def bar
    puts "my_hash  = #{my_hash}"
    my_hash = my_hash[:alpha]
    puts "my_hash  = #{my_hash}"
  end
end

Foo.new.bar
```
Running the above code will result in the following:
```
my_hash  = {:alpha=>"bravo"}
NoMethodError: undefined method `[]' for nil:NilClass
```

**Comment by afred** [comment]
hrm... i guess we don't want Travis authenticating with our AWS account. Not sure how we could do that securely. Maybe just not run it on Travis?

**Comment by afred** [comment]
@mccalluc Travis allows [encrypted environment variables](http://docs.travis-ci.com/user/environment-variables/). Worth looking into?

**Comment by afred** [comment]
@mccalluc ready for review and merge

**Comment by mccalluc** [comment]
@afred: Sorry, `params` was an astoundingly bad choice of method name. That said, I'm really not comfortable yet with the key stuff, though testing is a good thing. Let's get on the same page tomorrow?

**Comment by afred** [comment]
@mccalluc I created a new IAM user in AWS called **stock_sales_tester** and `.travis.yml` now uses those encrypted credentials

File: .travis.yml [hunk 1]
```diff
@@ -2,5 +2,9 @@ language: ruby
...
+  - secure: rpBvnyOEcpkuOPS/a6YGMDRla8zMi5WKcQYFoTSaNeChdKAUIiKxakcZsaAua2uqog6Wh+eS69/4vUJ1MDxUcKpMP5QosAC0VyKrTzPUrm464xeo2ZVYsyWVDV42gDhM2ahuWeQW9NS5YiUDbl9/UaeFPFYwUNYcNZqr/uVU2Sy7NSkvN/68CuMqztAmagqZkcCy7kI6tV2ogFd5g7DFb/NfMpx9YHmYNlhbzDOTvsHjDPW49FUoNpxPnT8K17+kYs2tZRdEyyjSjxS2h45qLTy2kGV3mfTzIGOw3+3Q75kEYSn/EgSaMuEX59uBHCLAc6xO3t7TznMEP/vBhTRcaAVK500MmKGJ9cUnqnnFg05VufrTnbm38ryqurjVmb6ks5GQdLWkb3Kl+zLsdJ7H8f9ZaBMhMtjZADMeHfpzfiOgM5TUVBTDPPtoVH6qAyBj2BieUP5ms2HX6CVNKZ41et9ZvmTkBVNeEB1E86fZzfJ2n05dq+7sUH1wHeN3m9jrOKk7wnPDfdvCgPEjBwGb41Lk1soev4nXgo9NJY/KkMycWoBJFasdENULDwZ65MEVM6+wAahv9luh8o4/GiIFEe3MzPckMx+uPFxHenC8WFAmBlrG42Y9cFkImCGccrnggg3fTNC/ZUoYR88B2zLTFSymMfvkMo1m5dLoSEsalWM=
```
Comment (mccalluc) [inline .travis.yml]: @afred: Do there really need to be two `secure`s? Could you provide a comment here about how these are (re)generated?

**Comment by mccalluc** [comment]
*Not* merge-able right now: Subclass is broken.

**Comment by afred** [comment]
@mccalluc can you revert 05345c3 and add tests for the subclass in a different PR?

**Comment by mccalluc** [comment]
Abandon this. Andrew's work was merged as part of #204. I'll file bugs for the remainder.

**Pull request closed without merging**